### PR TITLE
110 users can get more information on terms in the wiki

### PR DESCRIPTION
**Pull request by pascalalferink**

### CODE REVIEW

**Review by antw — COMMENTED** [review]
The code looks good to me, but I haven't been able to test it because I can't figure out how to create text that links to a StaticTermPage. 😕

I have a Static page and a StaticTermPage, and within the Static have tried writing text (that gets rendered by RawHtml) with the same name as the StaticTermPage. I tried both the value for the "Introduction" and the page title, and also tried linking to it directly but don't get any underlined text or a modal. Am I doing something wrong?

**Comment by mattijsstam** [comment]
> The code looks good to me, but I haven't been able to test it because I can't figure out how to create text that links to a StaticTermPage. 😕
> 
> I have a Static page and a StaticTermPage, and within the Static have tried writing text (that gets rendered by RawHtml) with the same name as the StaticTermPage. I tried both the value for the "Introduction" and the page title, and also tried linking to it directly but don't get any underlined text or a modal. Am I doing something wrong?

@antw you have to highlight text in the rich text editor and then you can replace the highlighted text with a term. I agree that the default interface is not intuitive

File: frontend/components/RawHtml/RawHtml.tsx [hunk 1]
```diff
@@ -1,8 +1,96 @@
...
+                    <div className="mt-2">
+                      <p className="text-sm text-gray-500">{modalText}</p>
+                    </div>
+                    <div className="mt-4">
+                      <HolonButton tag="a" href={modalLink} variant="darkmode">
+                        Lees meer
+                      </HolonButton>
+                    </div>
+                    <button
+                      onClick={closeModal}
```
Comment (antw) [inline frontend/components/RawHtml/RawHtml.tsx:83]: A button with no text content needs an [accessible label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) for screen readers.

```suggestion
                      onClick={closeModal}
                      aria-label="Sluiten"
```

File: frontend/components/RawHtml/RawHtml.tsx [hunk 2]
```diff
@@ -1,8 +1,96 @@
...
+      }
+    }
+  }, []);
+
+  return (
+    <Fragment>
+      <div ref={RawHtmlItem} className={s.Container} dangerouslySetInnerHTML={{ __html: html }} />
+      {modalText && (
+        <Transition appear show={isOpen} as={Fragment}>
+          <Dialog as="div" className="relative z-10" onClose={closeModal}>
```
Comment (antw) [inline frontend/components/RawHtml/RawHtml.tsx:48]: This z-index is too low. The header has a value of 50, and if the viewport is small this causes the modal to be overlapped by the header:

Screenshot 2022-11-17 at 09 39 45

```suggestion
          <Dialog as="div" className="relative z-50" onClose={closeModal}>
```

**Review by antw — CHANGES_REQUESTED** [review]
I was able to test the modal (thanks @mattijsstam!) and found two small issues. Other than those, this feature worked perfectly.

**Review by antw — APPROVED** [review]

File: frontend/components/RawHtml/RawHtml.tsx [hunk 3]
```diff
@@ -1,8 +1,96 @@
...
+    setIsOpen(false);
+    setModalText("");
+    setModalLink("");
+  }
+
+  function openModal(text, linkUrl) {
+    setModalLink(linkUrl);
+    setModalText(text);
+    setIsOpen(true);
+  }
```
[the remark this answers is not in the document]
Comment (pascalalferink) [reply frontend/components/RawHtml/RawHtml.tsx]: Hi @antw , I agree with you that 3 lines of setState is unnecessary . I refactored the code, while not implementing the reducer. Hence the three properties are always changed together, I merged the three states into one object (Modal) with three properties.

Is this OK with you?

[the remark this answers is not in the document]
Comment (antw) [reply frontend/components/RawHtml/RawHtml.tsx]: Thanks, I think that's much better; and simpler than using the reducer as well. :+1:

**Review by antw — DISMISSED** [review]

**Review by broekhuisg — APPROVED** [review]
Getest en het werkt. Echt super tof gedaan!

**Pull request merged**